extensions/wikia/PortableInfobox/services/Parser/Nodes/NodeImage.php

```diff
@@ -76,6 +97,40 @@ private function getImageAsTitleObject( $imageName ) {
 		return $title;
 	}
 
+	private function getGalleryItems( $imageData ) {
+		preg_match( '/.(UNIQ.*QINU)./U', $imageData, $galleryMarkers );
+		$galleryImages = array();
+
+		foreach ( $galleryMarkers as $marker ) {
+			$galleryWikitext = PortableInfoboxDataBag::getInstance()->getGallery( $marker );
+			$galleryLines = preg_split('/\r\n|\n|\r/', $galleryWikitext);
+
+			foreach ( $galleryLines as $galleryImage) {
```

Comment:
For now let's not spend more time on it - as we discussed - I'm moving forward with regex based implementation and creating a follow up ticket to come up later with "better" approach (we will discuss it more during retrospective).